interfaces/cython/cantera/ctml2yaml.py

```diff
@@ -0,0 +1,2682 @@
+                    else:
+                        if spec.attribs["electrolyte-species-type"] != etype:
+                            warnings.warn(
+                                "The electrolyteSpeciesType node was specified at the "
+                                "phase and species level for species '{}'. The value "
+                                "specified in the species node will be "
+                                "used".format(name)
+                            )
+
+        return activity_data
+
+    @classmethod
+    def to_yaml(cls, representer, data):
+        """Serialize the class instance to YAML format suitable for ruamel.yaml.
+
+        :param representer:
+            An instance of a ruamel.yaml representer type.
+        :param data:
+            An instance of this class that will be serialized.
+
+        The class instance should have an instance attribute called ``attribs`` which
+        is a dictionary representing the information about the instance. The dictionary
+        is serialized using the ``represent_dict`` method of the ``representer``.
+        """
+        return representer.represent_dict(data.attribs)
+
+
+class SpeciesThermo:
+    def __init__(self, thermo: etree.Element) -> None:
+        """Represent the polynomial-type thermodynamic data for a `Species`.
+
+        :param thermo:
+            A ``species/thermo`` XML node. Must have one or more child nodes with tag
+            ``NASA``, ``NASA9``, ``const_cp``, ``Shomate``, or ``Mu0``.
+
+        This class will process the `Species`-level thermodynamic information for the
+        polynomial thermo types. The pressure-dependent standard state types are
+        processed directly in the `Species` instance.
+        """
+        thermo_type = thermo[0].tag
+        if thermo_type not in ["NASA", "NASA9", "const_cp", "Shomate", "Mu0"]:
+            raise TypeError("Unknown thermo model type: '{}'".format(thermo[0].tag))
+        func = getattr(self, thermo_type)
+        self.attribs = func(thermo)
+
+    def process_polynomial(
+        self, thermo: etree.Element, poly_type: str
+    ) -> Tuple[List[List[float]], List[float]]:
+        """Process the `Species` thermodynamic polynomial for several types.
+
+        :param thermo:
+            A ``species/thermo`` XML node. Must have one or more child nodes with tag
+            ``NASA``, ``NASA9``, or ``Shomate``.
+        :param poly_type:
+            A string determining the type of polynomial. One of ``NASA``, ``NASA9``,
+            or ``Shomate``.
+
+        This method converts the polynomial data for the ``NASA``, ``NASA9``, and
+        ``Shomate`` thermodynamic types into the appropriate YAML structure.
+        """
+        temperature_ranges = set()
+        model_nodes = thermo.findall(poly_type)
+        unsorted_data = {}
+        for node in model_nodes:
+            Tmin = float(node.get("Tmin", 0))
+            Tmax = float(node.get("Tmax
```

Comment:
I agree that there should be a consistent approach for calling converters. Here is another example that omits arguments altogether:
```
docker@xps13:/src$ python interfaces/cython/cantera/ck2yaml.py 

ck2yaml.py: Convert Chemkin-format mechanisms to Cantera YAML input files

Usage:
    ck2yaml [--input=<filename>]
            [--thermo=<filename>]
            [--transport=<filename>]
            [--surface=<filename>]
            [--name=<name>]
            [--output=<filename>]
            [--permissive]
            [-d | --debug]

Example:
    ck2yaml --input=chem.inp --thermo=therm.dat --transport=tran.dat

If the output file name is not given, an output file with the same name as the
input file, with the extension changed to '.yaml'.

An input file containing only species definitions (which can be referenced from
phase definitions in other input files) can be created by specifying only a
thermo file.

For the case of a surface mechanism, the gas phase input file should be
specified as 'input' and the surface phase input file should be specified as
'surface'.

The '--permissive' option allows certain recoverable parsing errors (e.g.
duplicate transport data) to be ignored. The '--name=<name>' option
is used to override default phase names (i.e. 'gas').
```
compared to
```
docker@xps13:/src$ python interfaces/cython/cantera/ctml2yaml.py 
Traceback (most recent call last):
  File "interfaces/cython/cantera/ctml2yaml.py", line 2188, in <module>
    convert(sys.argv[1], sys.argv[2])
IndexError: list index out of range

```

Reply:
@ischoegl that looks like an older version of the script before I added the CLI. Nonetheless, I agree that making all three YAML conversion scripts have a consistent interface would be a good idea though.

Reply:
If that's the error you're getting, I think you haven't updated to the most recent commit for this PR. The output I get is:
```
usage: ctml2yaml.py [-h] -i INPUT [-o OUTPUT] [--permissive]
ctml2yaml.py: error: the following arguments are required: -i/--input
```

Reply:
Beaten to the punch 😁. Consistency is nice, but I think it's important to recognize that `ck2yaml`, like `ck2cti` before it, has a lot more variety in its possible usage compared to `cti2yaml` and `ctml2yaml`.

Reply:
By consistency, I also meant using `argparse` for all of them. Considering we probably want `ck2yaml` to be similar to `ck2cti` in terms of the interface, we could maybe leave that one alone, but @speth if you agree I'll add an `argparse` interface to `cti2yaml`. I actually added the `--input` here to better match `ck2yaml`. I looked for a way to have it be either positional or an option but that apparently isn't supported.

Reply:
Yes, a better interface for `cti2yaml` would we welcome. What's there just survived as a slight improvement over what it inherited from `ctml_writer`. 

I don't really see a benefit to replacing the `getopt` parsing done in `ck2yaml` with `argparse`.

Reply:
While I added some comments below as well, my point is that it would be nice if all converters had a similar ‘look and feel’, regardless of heritage. I personally like the more verbose output of the older versions. Ultimately, `ck2yaml` and `ctml2yaml` are the ones that survive and should be consistent. Both are new in 2.5, so a break of style compared to `ck2cti` is legitimate.

Reply:
@speth Since your original suggestion was to make `ctml2yaml.py` more like `cti2yaml.py`, do you agree with going the other way (make `cti2yaml.py` more like `ctml2yaml.py`)? That would make both scripts more similar to the way that `ck2cti.py` and `ck2yaml.py` work in terms of the interface.

Reply:
I don't really like the idea of having to use long options for a script whose basic usage could follow the pattern of standard commands like `cp`, but if you think there's more value in consistency with `ck2yaml`, then that's fine.

Reply:
@speth I can see that perspective, particularly now that the optional `--permissive` argument is removed from `ctml2yaml.py`. I tried to have both options available (positional and long options), but that just seemed needlessly confusing. I landed on just having the positional options, and added some `epilog` text as @ischoegl suggested.